### ISSUE TITLE
please port to python3

### ISSUE THREAD

**Issue by LocutusOfBorg** [jmbr/halberd#2]
Hi, I'm going to package this really useful tool for Debian, would it possible to port it to python3
thanks

**Comment by jmbr** [comment]
Hi, thanks for your interest.  This is certainly possible but I cannot give you a concrete timeline for it yet.

**Comment by LocutusOfBorg** [comment]
Hi, you might want to use 2to3 to port it to python3 and 3to2 at build time to build the old package...
(or vice-versa)... do you think it is possible?

thanks

**Comment by jmbr** [comment]
Yes, this appears to be reasonably straight-forward.  I'll get back to 
you soon.

On 08/07/2015 02:03 PM, Gianfranco Costamagna wrote:
> Hi, you might want to use 2to3 to port it to python3 and 3to2 at build
> time to build the old package...
> (or vice-versa)... do you think it is possible?
>
> thanks
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jmbr/halberd/issues/2#issuecomment-128796381>.
>

-- 
Juan M. Bello-Rivas

**Comment by LocutusOfBorg** [comment]
Hint: using 3to2 means probably just adding something like line 96 to your setup.py file
https://github.com/PyGithub/PyGithub/blob/master/setup.py

**Comment by BrentonEarl** [comment]
I package halberd for Slackware Linux.  I am also interested in providing python 3 support.  Any status on this?

**Comment by jmbr** [comment]
Thank you for your interest and for packaging Halberd.

I currently don't have time to do active development on the program but I'd be willing to review and merge a patch or pull request that provides Python 3 support.

**Comment by BrentonEarl** [comment]
Ok good to know.  I will add it to my to do list.  Thanks for your quick response!

**Comment by fabaff** [comment]
Python 2 will be EOL soon.

To be able to ship `halberd` in the future with Fedora support for Python 3 is needed. I guess that this is the case for all other distributions too.

**Comment by LocutusOfBorg** [comment]
Debian will drop it too if it doesn't get ported to Python3 before next stable (~2 years)

**Comment by LocutusOfBorg** [comment]
Hello, the package is getting autoremoved from Debian testing in some days...

**Comment by fabaff** [comment]
Same for Fedora...
